go/fn/examples/example_asmain_runner_test.go

```diff
@@ -19,43 +19,108 @@ import (
 	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
 )
 
+var _ fn.Runner = &CustomFnConfig{}
+
+type CustomFnConfig struct {
+	Owner string
+	Org   string
+}
+
+// Run is the main function logic.
+// `ctx` provides easy methods to add info, error or warning result to `ResourceList.Results`.
+// `items` is parsed from the STDIN "ResourceList.Items".
+// `functionConfig` is from the STDIN "ResourceList.FunctionConfig". The value is parsed from a `CustomFnConfig` type
+// "owner" and "org" field.
+func (r *CustomFnConfig) Run(ctx *fn.Context, functionConfig *fn.KubeObject, items fn.KubeObjects) {
+	for _, o := range items {
+		o.SetName(r.Owner)
+		o.SetNamespace(r.Org)
+	}
+	ctx.ResultInfo("updated namespace and name", nil)
+}
+
+// This example uses a CustomFnConfig object, which implements `Runner.Run` methods.
+func Example_asMainCustomFnConfig() {
+	file, _ := os.Open("./data/runner-customFnConfig.yaml")
```

Comment:
I'd move these to golden tests, because that keeps the function itself cleaner and shows how to test!